ostd/src/lib.rs

```diff
@@ -75,14 +75,15 @@ unsafe fn init() {
     #[cfg(feature = "cvm_guest")]
     arch::init_cvm_guest();
 
+    logger::init();
+
     // SAFETY: This function is called only once and only on the BSP.
     unsafe { cpu::local::early_init_bsp_local_base() };
 
     // SAFETY: This function is called only once and only on the BSP.
     unsafe { mm::heap_allocator::init() };
 
-    boot::init();
-    logger::init();
+    boot::copy_boot_context_info_to_heap();
```

Comment:
This `copy_boot_context_info_to_heap` function exposes too much details to the top-level initialization flow, violating the principles of encapsulation and information hiding. From the highest level, it suffices to know that it is an initialization step after the heap is ready. So I suggest calling `boot::init_after_heap`.